src/rdkafka_cgrp.c

```diff
@@ -2434,6 +2435,280 @@ static void rd_kafka_cgrp_offset_commit_tmr_cb (rd_kafka_timers_t *rkts,
+        }
+
+        rd_kafka_cgrp_version_new_barrier(rkcg);
+
+        rd_kafka_dbg(rkcg->rkcg_rk, CGRP|RD_KAFKA_DBG_CONSUMER, "ASSIGN",
+                "Group \"%s\": adding %d partitions to existing assignment "
+                "of %d partitions in join state %s (v%"PRId32")",
+                rkcg->rkcg_group_id->str, partitions->cnt,
+                !rkcg->rkcg_assignment ? 0 : rkcg->rkcg_assignment->cnt,
+                rd_kafka_cgrp_join_state_names[rkcg->rkcg_join_state],
+                rkcg->rkcg_version);
+
+        if (!rkcg->rkcg_assignment)
+                rkcg->rkcg_assignment = rd_kafka_topic_partition_list_new(
+                                                partitions->cnt);
+
+        /* Add and mark partition(s) as desired */
+        for (i = 0; i < partitions->cnt; i++) {
+                rd_kafka_toppar_t *rktp;
+                const rd_kafka_topic_partition_t *rktpar =
+                        &partitions->elems[i];
+
+                /* This marks the partition as desired if it's not already. */
+                rktp = rd_kafka_toppar_get2(rkcg->rkcg_rk,
+                                            rktpar->topic,
+                                            rktpar->partition,
+                                            0/*no-ua*/, 1/*create-on-miss*/);
+
+                if (!rktp)
+                        /* LOG_ERR already emitted by rd_kafka_toppar_get2 */
+                        continue;
+
+                rd_kafka_toppar_keep(rktp);
+
+                partitions->elems[i]._private = rktp;
+                if (!rd_kafka_topic_partition_list_add0(
+                          rkcg->rkcg_assignment,
+                          rktpar->topic,
+                          rktpar->partition,
+                          rktp)) {
+                        rd_assert(
+                                !*"rd_kafka_topic_partition_list_add0 failed");
+                }
+        }
+
+        rd_kafka_wrlock(rkcg->rkcg_rk);
+        rkcg->rkcg_c.assignment_size += partitions->cnt;
+        rd_kafka_wrunlock(rkcg->rkcg_rk);
+
+        rd_kafka_cgrp_set_join_state(rkcg,
+                                     RD_KAFKA_CGRP_JOIN_STATE_ASSIGNED);
+
+        if (RD_KAFKA_CGRP_CAN_FETCH_START(rkcg))
+                rd_kafka_cgrp_partitions_fetch_start(
+                        rkcg, partitions, 0);
+
+        return NULL;
+}
+
+
+/**
+ * @brief Incrementally remove from an existing partition assignment
+ *        May update \p partitions but will not hold on to it.
+ *
+ * @returns an error object or NULL on success.
+ */
+static rd_kafka_error_t *
+rd_kafka_cgrp_incremental_unassign (rd_kafka_cgrp_t *rkcg,
+                                    rd_kafka_topic_partition_list_t
+                                    *partitions) {
+        int i;
+        rd_kafka_resp_err_t err;
+        int cur_assignment_cnt =
+                (rkcg->rkcg_assignment ? rkcg->rkcg_assignment->cnt : 0);
+
+        rd_assert(partitions);
```

Comment:
There's no need to assert on this unless you expect the callers to be buggy, but since the number of callers is miniscule that is unlikely.

Reply:
ok, i'll remove it, it provides only tiny value. still, i don't think it hurts to have these things littered about. if you have 100 of them, 1 of them will probably end up paying off some time.